app/scripts/modules/amazon/src/loadBalancer/configure/application/ALBListeners.tsx

```diff
@@ -328,6 +328,26 @@ export class ALBListeners extends React.Component<IALBListenersProps, IALBListen
     this.updateListeners();
   };
 
+  private handleHttpRequestMethodChanged = (
+    condition: IListenerRuleCondition,
+    newValue: string,
+    selected: boolean,
+  ): void => {
+    let newValues = condition.values || [];
+
+    if (selected) {
+      newValues.push(newValue);
+    } else {
+      newValues = newValues.filter(v => v !== newValue);
+    }
+
+    condition.values = newValues;
```

Comment:
What's the reason to duplicate the array in `values` and `httpRequestMethodConfig.values`?

Reply:
For this condition type, `httpRequestMethodConfig` is the source of truth from aws, and then I transform it to `values` to match the model of the other conditions in the UI. We don't really use `httpRequestMethodConfig` in the UI (except in the LB transform), but  I just wanted to keep the values consistent.

Reply:
will you add that as a comment in the source code?  The way I see it, this is the kind of tech debt that is impossible to pay off because future me doesn't remember if the fields are used or not.

Reply:
The alternative would be to transform both ways (when sending and receiving), but I'm not sure if that's feasible here.

Reply:
Yes I will add a comment